### PR TITLE
Add guide about how to set hive.metastore.warehouse.dir

### PR DESCRIPTION
**Pull request by waterlx**
For issue #436

### CODE REVIEW

**Comment by waterlx** [comment]
`hive.metastore.warehouse.dir` is supposed to set in hive-site.xml, in common sense. But SparkContext#hadoopConfiguration() does not take hive-site.xml into account, so even though you might copy hive-site.xml into $HADOOP_CONF_DIR, it does not work in Spark.
Setting it in the code is the most explicit way to configure it.

**Comment by waterlx** [comment]
@rdblue does it make any sense to you?

File: site/docs/api-quickstart.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ The Hive catalog connects to a Hive MetaStore to keep track of Iceberg tables. T
 ```scala
 import org.apache.iceberg.hive.HiveCatalog
 
+spark.sparkContext.hadoopConfiguration.set("hive.metastore.warehouse.dir", "path/to/warehouse")
```
Comment (rdblue) [inline site/docs/api-quickstart.md]: The temporary path should be a real one. Let's use `s3://bucket/path/to/warehouse`.

Also, can you add a comment about being able to configure Hive here and add a newline between this and instantiating the catalog?

**Comment by waterlx** [comment]
Close this PR by mistake (or github has something wrong?), trying to re-open it, but failed

File: site/docs/api-quickstart.md (hunk 1 above)
Comment (waterlx) [reply site/docs/api-quickstart.md]: Thanks very much for the comment! Addressed.

**Pull request closed without merging**